### PR TITLE
chore: updated to SDK v0.50.6 and added the missing code

### PR DESCRIPTION
**Pull request by vishal-kanna**

### CODE REVIEW

**Comment by taariq** [comment]
@vishal-kanna will you resolve the merge conflicts here?

**Comment by byte-bandit** [comment]
Needs a rebase

File: go.mod [hunk 1]
```diff
@@ -263,8 +265,6 @@ replace (
 	// in order to try and avoid consensus hash mismatches.
 	// See: https://github.com/palomachain/paloma/issues/1144
 	// See https://github.com/palomachain/paloma/pull/1139
-	cosmossdk.io/store v1.0.2 => cosmossdk.io/store v1.1.0
-	github.com/cosmos/iavl => github.com/cosmos/iavl v1.1.2
-	github.com/prometheus/client_golang => github.com/prometheus/client_golang v1.18.0
-	github.com/prometheus/common => github.com/prometheus/common v0.47.0
+	github.com/prometheus/client_golang => github.com/prometheus/client_golang v1.19.0
+	github.com/prometheus/common => github.com/prometheus/common v0.52.2
```
Comment (byte-bandit) [inline go.mod:269]: These can both be removed I think, Cosmos `0.50.6` pulls them in as well.
Meaning we can hopefully remove the entire `replace` block here.

Comment (taariq) [reply go.mod]: resolving this.

**Comment by maharifu** [comment]
This will now be handled in https://github.com/palomachain/paloma/pull/1173

**Pull request closed without merging**